lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs

```diff
@@ -90,8 +102,59 @@ spec = do
         it "throws when length (key) != 32" $
             evaluate (singleAddressFromKey (Proxy @'Mainnet) (XPub "\148" cc))
             `shouldThrow` anyErrorCall
+    describe "keyToAddress is consistent with jcli" $ do
+        -- The goldens have been verified with jcli as follows:
+        -- $ jcli key generate --type=Ed25519 | jcli key to-public
+        -- ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c
+        -- $ jcli address single ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c --testing
+        -- ta1sv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4wwywkd6j
+        -- $ jcli address single ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c
+        -- ca1qv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4ww0xfchg
+
+        bech32KeyToAddrGolden
+            "ed25519_pk1yv0er3wlzvcauqj470tesdlcwy4dll9w7cvsvn4q30678gh44tnsxdh75c"
+            "ta1sv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4wwywkd6j"
+            "ca1qv33lyw9mufnrhsz2hea0xphlpcj4hlu4mmpjpjw5z9ltcaz7k4ww0xfchg"
+
+        bech32KeyToAddrGolden
+            "ed25519_pk1gf85kgfspqzgz9mvmral4e8zatg6d64ep3gk223fla09parjm7zqf4xy3n"
+            "ta1sdpy7jepxqyqfqghdnv0h7hyut4drfh2hyx9zef298l4u585wt0cgmkm96t"
+            "ca1qdpy7jepxqyqfqghdnv0h7hyut4drfh2hyx9zef298l4u585wt0cgs7ysh3"
+
+        bech32KeyToAddrGolden
+            "ed25519_pk16cwyfh3jy6ls8ypfpen8pk55d3ufkp9793xc96u32rl83uuv89nsxsnnhg"
+            "ta1s0tpc3x7xgnt7qus9y8xvux6j3k83xcyhckymqhtj9g0u78n3sukwk7ku9u"
+            "ca1q0tpc3x7xgnt7qus9y8xvux6j3k83xcyhckymqhtj9g0u78n3sukwakffgx"
+
+bech32KeyToAddrGolden
+    :: Text
+    -- ^ public key ("ed25519_pk...")
+    -> Text
+    -- ^ corresponding testnet address ("ta...")
+    -> Text
+    -- ^ corresponding mainnet address ("ca...")
+    -> Spec
```

Comment:
Reasoning for using `Text -> Text -> Text`: 
- The human readable parts of the bech32 strings already act as types (just not at compile-time)
- Allows for easy copy-and-paste at call-site for comparison with `jcli`